pkg/webmiddleware/proxy_headers.go

```diff
@@ -0,0 +1,122 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webmiddleware
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+const (
+	headerForwarded = "Forwarded"
+
+	headerXForwardedFor   = "X-Forwarded-For"
+	headerXForwardedHost  = "X-Forwarded-Host"
+	headerXForwardedProto = "X-Forwarded-Proto" // We don't support non-standard headers such as Front-End-Https, X-Forwarded-Ssl, X-Url-Scheme.
+	headerXRealIP         = "X-Real-IP"
+
+	schemeHTTP  = "http"
+	schemeHTTPS = "https"
+)
+
+var (
+	proxyHeaders        = []string{headerForwarded, headerXForwardedFor, headerXForwardedHost, headerXForwardedProto, headerXRealIP}
```

Comment:
Consider adding `X-Forwarded-Client-Cert` for mTLS

Reply:
Ok. I added the headers injected by tls-terminating reverse proxies Envoy/HAProxy and Traefik. In `nginx` the header is configurable.

Similar to what we (don't) do with the Forwarded/XFF header, we'll leave it to handler implementations to detect the header and parse it.